libs/git_wrapper/Git_wrapper.ml

```diff
@@ -208,6 +228,42 @@ let get_project_root_exn () =
   | Some path -> path
   | _ -> raise (Error "Could not get git root from git rev-parse")
 
+let checkout ?cwd ?git_ref () =
+  let cmd = (git, cd cwd @ [ "checkout" ] @ opt git_ref) in
+  match UCmd.status_of_run cmd with
+  | Ok (`Exited 0) -> ()
+  | _ -> raise (Error "Could not checkout git ref")
+
+let sparse_shallow_filtered_checkout url path =
+  let path = Fpath.to_string path in
+  let cmd =
+    ( git,
+      [
+        "clone";
+        "--depth=1";
+        "--filter=blob:none";
```

Comment:
maybe some comments about why the use of those particular options,
with maybe in comment the discussion about speed and your microbench experiment.

Reply:
As this module evolves, we'll probably want to split it into generic functions with many options and specialized functions such as `sparse_shallow_filtered_checkout` that combine many options internally.

Reply:
added comments